drivers/soft_spi/soft_spi.c

```diff
@@ -137,29 +138,33 @@ void soft_spi_release(soft_spi_t bus)
 
 static inline uint8_t _transfer_one_byte(soft_spi_t bus, uint8_t out)
 {
-    int8_t bit = 0, i = 0;
+    int8_t i = 8;
```

Comment:
```suggestion
    uint8_t i = 8;
```

It is better to use `unsigned` types for bit operations, as C allows for [arithmetic shifts](https://en.wikipedia.org/wiki/Arithmetic_shift) to be used on signed types.

Reply:
I'll take your advice